### PR TITLE
Integrate argument inference

### PR DESCRIPTION
**Pull request by RobbeSneyders**
Follow-up to use the argument inference functionality added in #763

### CODE REVIEW

File: src/fondant/pipeline/lightweight_component.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def __post_init__(self):
             self.base_image = "fondant:latest"
 
 
-class PythonComponent:
+class PythonComponent(BaseComponent):
```
Comment (RobbeSneyders) [inline src/fondant/pipeline/lightweight_component.py:23]: Needed this to satisfy typing.

File: tests/pipeline/test_python_component.py [hunk 2]
```diff
@@ -85,4 +93,7 @@ def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
         ref=AddN,
         produces={"x": pa.int32(), "y": pa.int32()},
         consumes={"x": pa.int32(), "y": pa.int32()},
+        arguments={"n": 1},
     )
+
+    DockerCompiler().compile(pipeline)
```
Comment (RobbeSneyders) [inline tests/pipeline/test_python_component.py:99]: This expands the coverage of this test by including the compilation without errors of lightweight components.

File: src/fondant/pipeline/lightweight_component.py [hunk 3]
```diff
@@ -42,12 +42,12 @@ def wrapper(cls):
 
         # updated=() is needed to prevent an attempt to update the class's __dict__
         @wraps(cls, updated=())
-        class AppliedPythonComponent(cls, PythonComponent):
+        class PythonComponentOp(cls, PythonComponent):
```
Comment (RobbeSneyders) [inline src/fondant/pipeline/lightweight_component.py:45]: The component is not applied yet at this point.

File: examples/sample_pipeline/pipeline.py [hunk 4]
```diff
@@ -50,13 +50,17 @@
...
-        dataframe["chunk_length"] = dataframe["text"].apply(len)
+        dataframe[self.feature_name] = dataframe["text"].apply(len)
         return dataframe
 
 
 _ = dataset.apply(
     ref=CalculateChunkLength,
     consumes={"text": pa.string()},
     produces={"chunk_length": pa.int32()},
+    arguments={"feature_name": "chunk_length"},
```
Comment (PhilippeMoussalli) [inline examples/sample_pipeline/pipeline.py:65]: This is a bit confusing, can't we just keep it as it was before and pass a random string argument

File: src/fondant/pipeline/pipeline.py [hunk 5]
```diff
@@ -222,6 +223,7 @@ def from_ref(cls, ref: t.Any, **kwargs) -> "ComponentOp":
                     description=description,
                     consumes={"additionalProperties": True},
                     produces={"additionalProperties": True},
+                    args={k: v.to_spec() for k, v in infer_arguments(ref).items()},
```
Comment (PhilippeMoussalli) [inline src/fondant/pipeline/pipeline.py:226]:
```suggestion
                    args={arg_name: arg.to_spec() for arg_name, arg  in infer_arguments(ref).items()},
```

just for clarity

**Review by PhilippeMoussalli — APPROVED** [review]
Thanks Robbe!

File: examples/sample_pipeline/pipeline.py (hunk 4 above)
Comment (mrchtr) [inline examples/sample_pipeline/pipeline.py:65]: nitpick: looks on a first view a bit confusing. Could we pass an dummy argument which is not used and keep `dataframe["chunk_length"]`?

**Review by mrchtr — APPROVED** [review]

File: examples/sample_pipeline/pipeline.py (hunk 4 above)
Comment (PhilippeMoussalli) [reply examples/sample_pipeline/pipeline.py:65]: great minds think alike

**Review by GeorgesLorre — APPROVED** [review]
no comment only approve

**Pull request merged**